src/main/java/net/earthcomputer/clientcommands/command/MinesweeperCommand.java

```diff
@@ -0,0 +1,341 @@
+    }
+
+    private void generateMines(int avoidX, int avoidY) {
+        Random random = new Random();
+
+        for (int i = 0; i < mines; i++) {
+            int x = random.nextInt(boardWidth);
+            int y = random.nextInt(boardHeight);
+
+            // too close to the clicked position
+            if (Mth.abs(avoidX - x) <= 1 && Mth.abs(avoidY - y) <= 1) {
+                i--;
+                continue;
+            }
+
+            // already a mine
+            if ((board[y * boardWidth + x] & 0b1100) >>> 2 == 2) {
+                i--;
+                continue;
+            }
+
+            incrementWarning(x - 1, y - 1);
+            incrementWarning(x, y - 1);
+            incrementWarning(x + 1, y - 1);
+
+            incrementWarning(x - 1, y);
+            board[y * boardWidth + x] = 0b000_10_0_0;
+            incrementWarning(x + 1, y);
+
+            incrementWarning(x - 1, y + 1);
+            incrementWarning(x, y + 1);
+            incrementWarning(x + 1, y + 1);
+        }
+    }
+
+    private void incrementWarning(int x, int y) {
+        if (isWithinBounds(x, y)) {
+            int idx = y * boardWidth + x;
+            byte original_tile = board[idx];
+            if ((original_tile & 0b1100) >>> 2 == 1) {
+                // increment warning quantity
+                board[idx] += 0b0_001_00_0_0;
+            } else if ((original_tile & 0b1100) >>> 2 == 0) {
+                // set to warning tile (and since it was empty beforehand, we make it have a quantity of 1)
+                board[idx] = 0b0_000_01_0_0;
+            }
+        }
+    }
+
+    private boolean isWithinBounds(int x, int y) {
+        return 0 <= x && x < boardWidth && 0 <= y && y < boardHeight;
+    }
+
+    private void click(int x, int y) {
+        byte tile = board[y * boardWidth + x];
+        boolean flagged = (tile & 2) > 0;
+        boolean covered = (tile & 1) == 0;
+        int type = (tile & 0b1100) >>> 2;
+        if (!covered || flagged) {
+            return;
+        }
+
+        if (type == 1) {
+            // set uncovered
+            board[y * boardWidth + x] |= 0b1;
+            emptyTilesRemaining -= 1;
+        } else if (type == 2) {
+            // set uncovered
+            board[y * boardWidth + x] |= 0b1;
+            deathCoords = new Vector2i(x, y);
+        } else {
+            // set uncovered
+            board[y * boardWidth + x] |= 0b1;
+            emptyTilesRemaining -= 1;
+            // we need to leave room for the current tile in the queue
+            int[] queue = new int[emptyTilesRemaining + 1];
+            int queueIdx = 0;
+            queue[0] = y * boardWidth + x;
+            while (queueIdx >= 0) {
+                int idx = queue[queueIdx--];
+                int x_part = idx % boardWidth;
+                int y_part = idx / boardWidth;
+                for (Vector2i possibleNeighbour : new Vector2i[]{
+                        new Vector2i(x_part - 1, y_part - 1),
```

Comment:
Please use camelCase for variables, there are a few occurrences of variables using snake_case.